src/fast-cpp-csv-parser/csv.h

```diff
@@ -849,17 +867,17 @@ namespace io{
                         char**sorted_col,
                         const std::vector<int>&col_order
                 ){
-                        for(std::size_t i=0; i<col_order.size(); ++i){
+                        for (int i : col_order) {
                                 if(line == nullptr)
                                         throw ::io::error::too_few_columns();
                                 char*col_begin, *col_end;
                                 chop_next_column<quote_policy>(line, col_begin, col_end);
 
-                                if(col_order[i] != -1){
+                                if (i != -1) {
                                         trim_policy::trim(col_begin, col_end);
                                         quote_policy::unescape(col_begin, col_end);
-                                                               
-                                        sorted_col[col_order[i]] = col_begin;
+
+                                        sorted_col[i] = col_begin;
```

Comment:
I'm not sure where col_order comes from, but to be safe we should bounds check it before using as an index.

Reply:
the change above is an extract of this upstream PR https://github.com/ben-strasser/fast-cpp-csv-parser/pull/88

the relevant part of changeset is the following:

```patch
@@ -865,17 +865,17 @@ namespace io{
                 void parse_line(
                         char*line,
                         char**sorted_col,
                         const std::vector<int>&col_order
                 ){
-                        for(std::size_t i=0; i<col_order.size(); ++i){
+                        for (int i : col_order) {
                                 if(line == nullptr)
                                         throw ::io::error::too_few_columns();
                                 char*col_begin, *col_end;
                                 chop_next_column<quote_policy>(line, col_begin, col_end);

-                                if(col_order[i] != -1){
+                                if (i != -1) {
                                         trim_policy::trim(col_begin, col_end);
                                         quote_policy::unescape(col_begin, col_end);

-                                        sorted_col[col_order[i]] = col_begin;
+                                        sorted_col[i] = col_begin;
                                 }
                         }
```